### PR TITLE
Allow customers to upload images on frontend

### PR DESCRIPTION
**Pull request by spaghetticode**
The new section on the frontend allows customers to upload images for
a given product. The new form is accessible from the product page via
link injected with `Deface`. When customer is logged in then the email
field is automatically populated with their email.
    
Integration and controller specs are added.

### CODE REVIEW

File: app/controllers/spree/customer_images_controller.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    before_action :load_product
+
+    def new
+      @customer_image = Spree::CustomerImage.new(base_review_params)
+    end
+
+    def create
+      @customer_image = Spree::CustomerImage.new(review_params)
+      if @customer_image.save
+        flash[:success] = Spree.t('customer_image_successfully_submitted')
```
Comment (kennyadsl) [inline app/controllers/spree/customer_images_controller.rb]: Better to use plain `t('spree.something')`.  [`Spree.t` is no more a good practice](https://github.com/solidusio/solidus/issues/2773)

File: app/views/spree/customer_images/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<%= content_tag :h2, Spree.t(:add_customer_image_for, name: @product.name), class: 'new-customer-image__title' %>
+
+<p><%= Spree.t 'moderation_message' %></p>
```
Comment (kennyadsl) [inline app/views/spree/customer_images/new.html.erb]: same here on `Spree.t`

File: app/controllers/spree/customer_images_controller.rb (hunk 1 above)
Comment (spaghetticode) [reply app/controllers/spree/customer_images_controller.rb]: 👍

File: app/views/spree/customer_images/new.html.erb (hunk 2 above)
Comment (spaghetticode) [reply app/views/spree/customer_images/new.html.erb]: 👍

**Pull request merged**